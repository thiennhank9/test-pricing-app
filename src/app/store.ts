import { configureStore, ThunkAction, Action } from '@reduxjs/toolkit';
import prices from '../features/prices/pricesSlice';

export const store = configureStore({
  reducer: {
    prices
  },
});

export type AppDispatch = typeof store.dispatch;
export type RootState = ReturnType<typeof store.getState>;
export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  Action<string>
>;
