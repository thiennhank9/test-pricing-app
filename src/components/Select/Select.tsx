import React from "react";
import "./Select.css";

interface SelectProps {
  value: string;
  options: Array<string>;
  onChange(event: React.ChangeEvent<HTMLSelectElement>): void;
}

function Select(props: SelectProps) {
  const { value, options, onChange } = props;

  return (
    <select onChange={onChange} value={value} className="select">
      {options.map((option) => (
        <option key={option} value={option}>
          {option}
        </option>
      ))}
    </select>
  );
}

export default Select;
