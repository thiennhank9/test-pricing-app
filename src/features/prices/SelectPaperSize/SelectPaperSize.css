.select-paper-size {
  width: 150px;
}

.row-button-apply {
  margin-top: 10px;
  display: flex;
  justify-content: flex-end;
}
