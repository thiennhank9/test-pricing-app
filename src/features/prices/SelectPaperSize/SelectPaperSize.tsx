import React, { useState } from "react";
import Select from "components/Select";
import Button from "components/Button";
import { useDispatch } from "react-redux";
import { initialState, selectPaperSize } from "../pricesSlice";
import "./SelectPaperSize.css";

const paperSizes = ["A4", "A5", "B4", "B5"];

function SelectPaperSize() {
  const dispatch = useDispatch();
  const [value, setValue] = useState(initialState.paperSize);
  const onChange = (e: React.ChangeEvent<HTMLSelectElement>) => setValue(e.target.value);
  const onClick = () => dispatch(selectPaperSize(value));

  return (
    <div className="select-paper-size">
      <h3>Select paper size</h3>
      <div>
        <Select value={value} options={paperSizes} onChange={onChange} />
      </div>
      <div className="row-button-apply">
        <Button onClick={onClick}>Apply</Button>
      </div>
    </div>
  );
}

export default SelectPaperSize;
