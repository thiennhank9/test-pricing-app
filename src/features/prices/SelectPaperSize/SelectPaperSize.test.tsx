import Select from "components/Select";
import { Provider } from "react-redux";
import TestRenderer from "react-test-renderer";
import SelectPaperSize from "./SelectPaperSize";
import { store } from "app/store";

describe("SelectPaperSize", () => {
  test("render select paper size", () => {
    const testRenderer = TestRenderer.create(
      <Provider store={store}>
        <SelectPaperSize />
      </Provider>
    );
    expect(testRenderer.toJSON()).toMatchSnapshot();
    const testInstance = testRenderer.root;
    const select = testInstance.findByType(Select);
    expect(select.props.options).toEqual(["A4", "A5", "B4", "B5"]);
  });
});
