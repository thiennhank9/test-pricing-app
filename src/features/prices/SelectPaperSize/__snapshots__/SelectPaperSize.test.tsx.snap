// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectPaperSize render select paper size 1`] = `
<div
  className="select-paper-size"
>
  <h3>
    Select paper size
  </h3>
  <div>
    <select
      className="select"
      onChange={[Function]}
      value="A4"
    >
      <option
        value="A4"
      >
        A4
      </option>
      <option
        value="A5"
      >
        A5
      </option>
      <option
        value="B4"
      >
        B4
      </option>
      <option
        value="B5"
      >
        B5
      </option>
    </select>
  </div>
  <div
    className="row-button-apply"
  >
    <button
      onClick={[Function]}
    >
      Apply
    </button>
  </div>
</div>
`;
